sr_robot_commander/src/sr_robot_commander/sr_robot_state_saver.py

```diff
@@ -115,15 +115,19 @@ def __init__(self, name, hand_or_arm="both", hand_h=False, save_target=False):
         if rospy.is_shutdown():
             exit(0)
 
-        rospy.loginfo(current_dict)
-        rs.joint_state = JointState()
-        rs.joint_state.name = current_dict.keys()
-        rs.joint_state.position = current_dict.values()
-        rospy.logwarn(rs)
-        self._save(self._name, robot_name, rs)
+        self.save_state(current_dict, robot_name)
 
     def _target_cb(self, data):
         self._mutex.acquire()
         for n, joint in enumerate(data.joint_names):
             self._target_values[joint] = data.desired.positions[n]
         self._mutex.release()
+
+    def save_state(self, current_dict, robot_name):
+        rs = RobotState()
+        rospy.loginfo(current_dict)
+        rs.joint_state = JointState()
+        rs.joint_state.name = current_dict.keys()
+        rs.joint_state.position = current_dict.values()
+        rospy.logwarn(rs)
```

Comment:
Is is really a "warning"? Looks more like "information" or "debug". Also would logging of JointState will be self not, maybe log prefix "JointSate: " + rs

Reply:
you're right. it isn't. it was like this in the previous code I should've changed this

Reply:
just rs is fine as it prints as:
 `joint_state: 
  header: 
    seq: 0
    stamp: 
      secs: 0
      nsecs:         0
    frame_id: ''
  name: [rh_FFJ1, rh_FFJ2, rh_FFJ3, rh_FFJ4, rh_THJ4, rh_THJ5, rh_THJ1, rh_THJ2, rh_THJ3,
  rh_LFJ2, rh_LFJ3, rh_LFJ1, rh_LFJ4, rh_LFJ5, rh_RFJ4, rh_RFJ1, rh_RFJ2, rh_RFJ3,
  rh_MFJ1, rh_MFJ3, rh_MFJ2, rh_MFJ4, rh_WRJ2, rh_WRJ1]
  position: [0.0003690871326975653, 0.09929020293835578, 0.30761233292307466, -0.004451969455385729, 0.9392589880666513, 0.0397158322452853, 0.0067776932103811305, 0.06636227463585787, 0.04648246055283778, 0.08446156967768026, 0.26019691544985246, 0.00019760850091454785, -0.14396959576273272, 0.5067272559937432, 0.004332903855831738, 0.00036504122331315614, 0.0994860676041407, 0.3077083502304445, -0.0004902973077589579, 0.3076921016296774, 0.09952685469396805, -0.004294016735936879, 0.03918514128779549, 0.48868965788423857]
  velocity: []
  effort: []
multi_dof_joint_state: 
  header: 
    seq: 0
    stamp: 
      secs: 0
      nsecs:         0
    frame_id: ''
  joint_names: []
  transforms: []
  twist: []
  wrench: []
attached_collision_objects: []
is_diff: False
`